### PR TITLE
Hlo fixes

### PR DESCRIPTION
**Pull request by draganmladjenovic**
Fixes //xla/service/gpu/tests/gpu_int4_test and //xla/tools/hlo_opt:* tests on ROCm.

### CODE REVIEW

File: xla/tools/hlo_opt/BUILD [hunk 1]
```diff
@@ -59,7 +62,6 @@ cc_library(
         "//xla/service:hlo_graph_dumper",
         "//xla/service:platform_util",
         "//xla/service/gpu:executable_proto_cc",
-        "//xla/stream_executor/cuda:cuda_platform_id",
```
Comment (akuegel) [inline xla/tools/hlo_opt/BUILD:62]: This dependency is actually needed, we have a header include in gpu_opt.cc.

**Review by akuegel — CHANGES_REQUESTED** [review]
Please improve the PR description a bit to describe what kind of fixes are done.

File: xla/tools/hlo_opt/BUILD (hunk 1 above)
Comment (draganmladjenovic) [reply xla/tools/hlo_opt/BUILD:62]: Won't having dependency on //xla/stream_executor/cuda:cuda_platform be enough?

**Review by tdanyluk — APPROVED** [review]

File: xla/tools/hlo_opt/BUILD (hunk 1 above)
Comment (akuegel) [reply xla/tools/hlo_opt/BUILD:62]: Yes, that works as well. Turns out the include was not used.

**Pull request closed without merging**